api/qan/collector.proto

```diff
@@ -2,56 +2,89 @@ syntax = "proto3";
 
 package qan;
 
-option go_package = "qan";
+option go_package = "qanpb";
 
 import "qan/qan.proto";
 
-// Agent service contains RPCs to implement communication mechanism with qan-agent.
-// Namely query metrics, query explain etc.
-service Agent {
-  // DataInterchange defines duplex communication between Query analytics API and Agent.
-  rpc DataInterchange(stream AgentMessage) returns (stream ApiMessage);
+// Collector service accepts data from pmm-agent (via pmm-managed).
+service Collector {
+  // Collect accepts data from pmm-agent (via pmm-managed).
+  rpc Collect(CollectRequest) returns (CollectResponse);
 }
 
-// AgentMessage contains several Metrics Bucket for one or more periods of time.
-message AgentMessage {
+message CollectRequest {
   repeated MetricsBucket metrics_bucket = 1;
 }
 
-// MetricsBucket is aggregated message created by qan-agent.
+// MetricsBucket is aggregated message created by pmm-agent.
 // Contains information about one query selected in defined way from query class in specific period of time.
 message MetricsBucket {
-  string queryid = 1; // md5 of digest_text/fingerprint.
-  string fingerprint = 2; // digest_text - query signature. Query without values.
+  // md5 of digest_text/fingerprint.
+  string queryid = 1;
+  // digest_text - query signature. Query without values.
+  string fingerprint = 2;
+  //
   // Dimension Group.
+  //
+
+  // Server.
   string d_server = 3;
-  string d_database = 4; // postgress database.
-  string d_schema = 5; // mysql database; postgress schema; mongo collection.
-  string d_username = 6; // client user name.
-  string d_client_host = 7; // client IP or hostname.
-  map<string, string> labels = 8; // Custom labels names:values.
-  string agent_uuid = 9; // Identifier of agent that collect and send metrics.
-  int64 period_start = 10; // Time when collection of bucket started.
-  uint32 period_length = 11; // Duration of bucket.
-  string metrics_source = 12; // Ex.: slowlog, perf schema, Mongo DB, Percona Server, etc.
-  string example = 13; // One of query example from set found in bucket.
+  // PostgreSQL database.
+  string d_database = 4;
+  // MySQL database; PostgreSQL schema; MongoDB collection.
+  // FIXME(askomorokhov) why collection?!
+  string d_schema = 5;
+  // client user name.
+  string d_username = 6;
+  // client IP or hostname.
+  string d_client_host = 7;
+  // Custom labels names:values.
+  map<string, string> labels = 8;
+  // Identifier of agent that collect and send metrics.
+  string agent_uuid = 9;
+  // Time when collection of bucket started.
+  // FIXME(askomorokhov) what unit? why not google.protobuf.Timestamp?
+  int64 period_start = 10;
```

Comment:
@askomorokhov ^

Reply:
This is unix timestamp in seconds. Considered this arguments: https://stackoverflow.com/questions/3574716/date-and-time-type-for-use-with-protobuf#3627777
I see more disadvantages of using `google.protobuf.Timestamp` here.